tools/wheel_resolver/src/wheel_tags/tags.py

```diff
@@ -0,0 +1,107 @@
+"""
+Some methods for wheel fetching and selection
+"""
+
+import os
+from third_party.python.packaging.utils import parse_wheel_filename
+import third_party.python.packaging.tags as tags
+import third_party.python.distlib.locators as locators
+
+
+def is_compatible(wheel, taglist):
+    """
+    Check whether the given python package is a wheel compatible with the
+    current platform and python interpreter.
+    Compatibility is based on https://www.python.org/dev/peps/pep-0425/
+    """
+    # Get the tag from the wheel we're checking
+    _, _, _, tag = parse_wheel_filename(wheel)
+
+    # taglist is a list of tags that we've got either from the user
+    # or we've auto-generated them for this system
+    if taglist is None:
+        taglist = tags.sys_tags()
+
+    for system_tag in taglist:
+        if system_tag in tag:
+            return True
+    return False
+
+
+def get_basename(url):
+    return os.path.basename(url)
+
+
+def is_wheel_file(url):
+    """ Return whether or not this is a .whl file """
+    basename = os.path.basename(url)
+    _, ext = os.path.splitext(basename)
+    return ext == '.whl'
+
+
+def generate_tags_from_all_archs(archs):
```

Comment:
Might be cleaner to have this just return `tags.sys_tags()` if `len(archs) == 0`. You could get rid of the checks from `get_url()` and `is_compatible()` then.